packages/core/src/fhir-deduplication/resources/encounter.ts

```diff
@@ -49,47 +50,78 @@ export function deduplicateEncounters(encounters: Encounter[]): DeduplicationRes
  * Approach:
  * 1 map, where the key is made of:
  * - date
- * - class.code
+ * - practitioner participant
  */
 export function groupSameEncounters(encounters: Encounter[]): {
   encountersMap: Map<string, Encounter>;
   refReplacementMap: Map<string, string>;
   danglingReferences: Set<string>;
 } {
-  const encountersMap = new Map<string, Encounter>();
+  const l1EncountersMap = new Map<string, string>();
+  const l2EncountersMap = new Map<string, Encounter>();
+
   const refReplacementMap = new Map<string, string>();
   const danglingReferences = new Set<string>();
 
-  function assignMostDescriptiveStatus(
-    master: Encounter,
-    existing: Encounter,
-    target: Encounter
-  ): Encounter {
-    master.status = pickMostDescriptiveStatus(statusRanking, existing.status, target.status);
-    return master;
-  }
-
   for (const encounter of encounters) {
     const datetime = getDateFromResource(encounter, "datetime");
-    // TODO: Improve the key. Just date is not sufficient.
-    if (datetime) {
-      const key = JSON.stringify({ datetime });
-      fillMaps(
-        encountersMap,
-        key,
-        encounter,
+
+    const practitionerRefsSet = new Set<string>();
+
+    const getterKeys: string[] = [];
+    const setterKeys: string[] = [];
+
+    encounter.participant?.forEach(participant => {
+      const ref = participant.individual?.reference;
+      if (ref) {
+        if (ref.includes("Practitioner")) {
+          practitionerRefsSet.add(ref);
+          return;
+        }
+      }
+    });
+
+    const practitionerRefs = Array.from(practitionerRefsSet).map(p => ({ practitioner: p }));
+
+    if (datetime && practitionerRefs.length > 0) {
+      const practitionerAndDateKeys = createKeysFromObjectArray({ datetime }, practitionerRefs);
+      setterKeys.push(...practitionerAndDateKeys);
+      getterKeys.push(...practitionerAndDateKeys);
+    }
+
+    if (datetime && practitionerRefs.length === 0) {
+      const dateKey = JSON.stringify({ datetime });
+      setterKeys.push(dateKey);
+      getterKeys.push(dateKey);
+    }
```

Comment:
NABD, but we have the same condition twice, one for each code block - could use if/else to avoid a change in one of the `if`s make both blocks executing.